keanu-python/keanu/context.py

```diff
@@ -122,6 +123,18 @@ def to_java_vertex_array(self, l: Collection[Any]) -> JavaArray:
         java_import(self.jvm_view(), "io.improbable.keanu.vertices.Vertex")
         return self.to_java_array(list(map(lambda x: x.unwrap(), l)), self.jvm_view().Vertex)
 
+    class JavaException:
+
+        def __init__(self, type: str, message: str):
+            self.type = type
+            self.message = message
+
+        def __repr__(self) -> str:
+            return "{}: {}".format(self.type, self.message)
+
+    def from_java_exception(self, e: Py4JJavaError) -> 'KeanuContext.JavaException':
+        return KeanuContext.JavaException(e.java_exception.getClass().getName(), e.java_exception.getMessage())
```

Comment:
I wonder if `JavaException` should be the one unwrapping the Java exception class? (i.e. take raw Java exception in its constructor)

Reply:
That's a good idea. In fact I realised it would be much neater if I implemented it as a JavaObjectWrapper. Please take another look!

Reply:
I also made it implement `Exception` so you can catch and rethrow Py4J's exception. See the tests.